rascaline-torch/src/calculator.cpp

```diff
@@ -288,3 +296,99 @@ metatensor_torch::TorchTensorMap rascaline_torch::register_autograd(
         return remove_other_gradients(precomputed, forward_gradients);
     }
 }
+
+
+// ========================================================================== //
+
+static void check_selection_type(
+    const torch::IValue& selection,
+    std::string field,
+    bool tensormap_ok
+) {
+    if (selection.isNone()) {
+        // all good
+    } else if (selection.isCustomClass()) {
+        // check if we have either a Labels or TensorMap
+        try {
+            selection.toCustomClass<metatensor_torch::LabelsHolder>();
+        } catch (const c10::Error&) {
+            if (tensormap_ok) {
+                try {
+                    selection.toCustomClass<metatensor_torch::TensorMapHolder>();
```

Comment:
not sure what is happening here and what exactly `tensormap_ok` is expressing. What confuses me is how you can make a Labels object and a TensorMap out of a selection object. Can you add some doc?